jplag.endToEndTesting/src/main/java/de/jplag/endToEndTesting/helper/JPlagTestSuiteHelper.java

```diff
@@ -0,0 +1,146 @@
+package de.jplag.endToEndTesting.helper;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+
+import java.util.Arrays;
+import java.util.List;
+
+import javax.annotation.processing.FilerException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import de.jplag.endToEndTesting.constants.Constant;
+import de.jplag.options.LanguageOption;
+import model.ResultJsonModel;
+import model.TestCaseModel;
+
+public class JPlagTestSuiteHelper {
+
+	private static final Logger logger = LoggerFactory.getLogger("EndToEndTesting");
+
+	private String[] resourceNames;
+	private String temporaryFolderPath;
+	private List<ResultJsonModel> resultModel;
+	private LanguageOption languageOption;
+
+	/**
+	 * Helper class for the endToEnd tests. In this class the necessary resources
+	 * are loaded, prepared and copied for the tests based on the passed parameters.
+	 * An instance of this class loads all necessary paths and properties for a test
+	 * run with the specified language
+	 * 
+	 * @param languageOption for loading language-specific resources
+	 * @throws Exception
+	 */
+	public JPlagTestSuiteHelper(LanguageOption languageOption) throws Exception {
+		this.languageOption = languageOption;
+		this.resourceNames = new File(Constant.BASE_PATH_TO_JAVA_RESOURCES_SORTALGO.toString()).list();
+		this.temporaryFolderPath = getTempFolderPath();
+
+		this.resultModel = JsonHelper
+				.getResultModelFromPath(Constant.BASE_PATH_TO_JAVA_RESULT_JSON.toAbsolutePath().toString());
+		logger.info(String.format("temp path at [%s]", this.temporaryFolderPath));
+	}
+
+	/**
+	 * creates all necessary folder paths and objects for a test run.
+	 * 
+	 * @param classNames
+	 * @return
+	 * @throws Exception
+	 */
```

Comment:
Please complete the java docs for public methods and classes

Reply:
done